### PR TITLE
[DoNotMerge] Azure Site Recovery - Replication cluster APIs

### PR DESCRIPTION
**Pull request by pradheepm**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by raosuhas** [comment]
> 	"/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters":{

please use spaces not tabs. this comment is for the whole PR 

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

**Comment by raosuhas** [comment]
>                },

puts should return 200\201 : 

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4743 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

**Comment by raosuhas** [comment]
> 	"A2AClusterUnplannedFailoverInput":{

spaces not tabs

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:8758 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

**Comment by raosuhas** [comment]
> 			},

please use ProxyResource from common types

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:22371 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/examples/ReplicationProtectionClusters_Create.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      "body": {
+		   "id":"/Subscriptions/c183865e-6077-46f2-a3b1-deb0f4f4650a/resourceGroups/resourceGroupPS1/providers/Microsoft.RecoveryServices/vaults/vault1/replicationFabrics/eastus/replicationProtectionContainers/eastus-container/replicationProtectionClusters/cluster12",
+		   "name":"cluster12",
+		   "type":"Microsoft.RecoveryServices/vaults/replicationFabrics/replicationProtectionContainers/replicationProtectionClusters",
+		   "properties":{
+			  "friendlyName":"cluster12",
+			  "protectionClusterType":null,
+			  "primaryFabricFriendlyName":"eastus",
+			  "primaryFabricProvider":"AzureFabric",
+			  "recoveryFabricFriendlyName":"centraluseuap",
```
Comment (raosuhas) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/examples/ReplicationProtectionClusters_Create.json:31]: >			 [](http://example.com/codeflow?start=0&length=3)

spaces not tabs .. please fix in examples files as well

**Comment by raosuhas** [comment]
>       "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}":{

Just a note here : 

[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc030-avoid-excessive-resource-type-nesting](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc030-avoid-excessive-resource-type-nesting)

ARM only officially supports 3 levels of nested resource types (i.e. {type1}/{name1}/{type2}/{name2}/{type3}/{name3}). If a resource type is nested further than 3 levels ARM will use the RP manifest details of the third resource type when validating and routing the request. Tracked resources (routing type "default") should not be used beyond the third level of nesting.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4611 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

**Comment by raosuhas** [comment]
>       "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/remove":{

How is this different from directly deleting the replicationProtectionCluster ? 

Avoid using POST actions to do CRUD operations on resources , you should be using PUT\PATCH\DELETE to do direct operations on resources



---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4902 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

**Comment by raosuhas** [comment]
>                      "$ref":"#/definitions/ReplicationProtectionCluster"

Typically for POST operations , the response should indicate the results of the action rather than the parent resource

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:5029 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

**Comment by raosuhas** [comment]
>                      "$ref":"#/definitions/ReplicationProtectionCluster"

same comment here about the POST response

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:5115 in 9859437. [](commit_id = 9859437fb823e0cf52e999a69fdbad878f7237ec, deletion_comment = False)

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json [hunk 2]
```diff
@@ -4545,6 +4545,594 @@
...
+               "Tracks the Replication protection cluster async operation.":{
+                  "$ref":"./examples/ReplicationProtectionClusters_GetOperationResults.json"
+               }
+            }
+         }
+      },
+      "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/remove":{
+         "post":{
+            "tags":[
+               "ReplicationProtectionClusters"
```
Comment (ayfathim) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4905]: We dont need this

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json [hunk 3]
```diff
@@ -9138,6 +9769,14 @@
         "managementId": {
           "description": "The management Id.",
           "type": "string"
+        },		
+		"protectionClusterId": {
```
Comment (ayfathim) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:9773]: Fix indentation

**Comment by pradheepm** [comment]
> > "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters":{
> 
> please use spaces not tabs. this comment is for the whole PR
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec). [](commit_id = [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec), deletion_comment = False)

Fixed the indentation in all the newly added files.

**Comment by pradheepm** [comment]
> 

We are not going to implement POST method for /remove, removed from swagger spec.

**Comment by pradheepm** [comment]
> > "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters":{
> 
> please use spaces not tabs. this comment is for the whole PR
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec). [](commit_id = [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec), deletion_comment = False)

Fixed the indent.

**Comment by pradheepm** [comment]
> > ```
> >   "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}":{
> > ```
> 
> Just a note here :
> 
> https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc030-avoid-excessive-resource-type-nesting
> 
> ARM only officially supports 3 levels of nested resource types (i.e. {type1}/{name1}/{type2}/{name2}/{type3}/{name3}). If a resource type is nested further than 3 levels ARM will use the RP manifest details of the third resource type when validating and routing the request. Tracked resources (routing type "default") should not be used beyond the third level of nesting.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4611 in [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec). [](commit_id = [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec), deletion_comment = False)

Our service is very old, and it was modeled to 4 Levels of nested resource types. Following the existing service model. To change the Model will be big change and breaking change.

**Comment by pradheepm** [comment]
> > ```
> >   "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/remove":{
> > ```
> 
> How is this different from directly deleting the replicationProtectionCluster ?
> 
> Avoid using POST actions to do CRUD operations on resources , you should be using PUT\PATCH\DELETE to do direct operations on resources
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4902 in [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec). [](commit_id = [9859437](https://github.com/Azure/azure-rest-api-specs/commit/9859437fb823e0cf52e999a69fdbad878f7237ec), deletion_comment = False)

Removed this API.

**Comment by mentat9** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters": {

Use camelcase "subscription". Comment applies to other new APIs as well.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters": {

Use a better name like vaultName. resourceName is too generic. Comment applies to other new APIs as well.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
>             "$ref": "#/parameters/ApiVersion"

Should get this by $ref'ing ApiVersionParameter definition from v4 common types. Likewise ResourceGroupName (ResourceGroupNameParameter), and SubscriptionId (SubscriptionIdParameter). This comment applies to the other new APIs as well.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4561 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
>               "$ref": "#/definitions/ServiceError"

Should use the ErrorResponse definition from v4 common types. Comment applies elsewhere in APIs being added as well.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4597 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/create": {

This should be modeled as PUT, not a create POST action. I think this will just fail in ARM with 404 anyway.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4818 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
>           "responses": {

Need 200 for final response

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4940 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
>           "responses": {

Need 200 for final response

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:5017 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by mentat9** [comment]
@pradheepm - I started ARM review, but was unable to finish by the end of my rotation. Next oncall is @TimLovellSmith who will continue next week.

**Comment by raosuhas** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/create": {

Not fixed ?

---
In reply to: [1374373948](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1374373948) [](http://example.com/codeflow?ancestors=1374373948,1374373948)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4818 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by raosuhas** [comment]
>           "responses": {

Not fixed

---
In reply to: [1374373961](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1374373961) [](http://example.com/codeflow?ancestors=1374373961,1374373961)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4940 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by raosuhas** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters": {

Not fixed. Please fix this 

---
In reply to: [1374373889](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1374373889) [](http://example.com/codeflow?ancestors=1374373889,1374373889)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by raosuhas** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/create": {

This would actually work fine in ARM since this is a Proxy resource , but this is still not the correct way to model this since the replicationProtectionContainers resource is modelled above without a PUT and here it seems like you are using a POST /create to actually create the resource. 

This is not recommended the CRUD operations should be done with the PUT\PATCH\DELETE verbs not using POSTS

---
In reply to: [1387730962](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1387730962) [](http://example.com/codeflow?ancestors=1387730962,1374373948)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4818 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by pradheepm** [comment]
> > ```
> > "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/create": {
> > ```
> 
> This would actually work fine in ARM since this is a Proxy resource , but this is still not the correct way to model this since the replicationProtectionClusters resource is modelled above without a PUT and here it seems like you are using a POST /create to actually create the resource.
> 
> This is not recommended the CRUD operations should be done with the PUT\PATCH\DELETE verbs not using POSTS
> 
> In reply to: [1387730962](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1387730962)
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4818 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

Fixed.

**Comment by pradheepm** [comment]
> > ```
> > "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters": {
> > ```
> 
> Not fixed. Please fix this
> 
> In reply to: [1374373889](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1374373889)
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

Fixed

**Comment by pradheepm** [comment]
> > ```
> >       "responses": {
> > ```
> 
> Need 200 for final response
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4940 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

Fixed

**Comment by pradheepm** [comment]
> > ```
> >       "responses": {
> > ```
> 
> Need 200 for final response
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4940 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

Fixed

**Comment by pradheepm** [comment]
> > ```
> >       "responses": {
> > ```
> 
> Need 200 for final response
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:5017 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

Fixed

**Comment by pradheepm** [comment]
> > ```
> > "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters/{replicationProtectionClusterName}/create": {
> > ```
> 
> This should be modeled as PUT, not a create POST action. I think this will just fail in ARM with 404 anyway.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4818 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

Moved to PUT method

**Comment by pradheepm** [comment]
> > ```
> > "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters": {
> > ```
> 
> Use a better name like vaultName. resourceName is too generic. Comment applies to other new APIs as well.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66). [](commit_id = [5504713](https://github.com/Azure/azure-rest-api-specs/commit/55047137a9865a47187a92b0ad3f3f56a6268a66), deletion_comment = False)

We are using Resource name for all the old APIs.

**Comment by sjanamma** [comment]
>     "/Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectionClusters": {

any update on this @pradheepm?

---
In reply to: [1374373892](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1374373892) [](http://example.com/codeflow?ancestors=1374373892,1374373892)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4548 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by sjanamma** [comment]
JSON parsing issue because of an extra '}' at line 8638. That could be the reason why checks are failing.

**Comment by sjanamma** [comment]
>             "$ref": "#/parameters/ApiVersion"

Are you planning to fix this?

---
In reply to: [1374373904](https://github.com/Azure/azure-rest-api-specs/pull/21951#issuecomment-1374373904) [](http://example.com/codeflow?ancestors=1374373904,1374373904)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4561 in 5504713. [](commit_id = 55047137a9865a47187a92b0ad3f3f56a6268a66, deletion_comment = False)

**Comment by sjanamma** [comment]
>               "$ref": "#/definitions/CreateReplicationProtectionClusterInput"

PUT Request and Response must have the same schema. This is a blocking issue

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/put-request-response-scheme.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/put-request-response-scheme.md)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:4730 in cafe77b. [](commit_id = cafe77bd13ae3b602ca022365071a1b9c55b7be4, deletion_comment = False)

**Pull request closed without merging**